wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -348,27 +348,26 @@ public boolean getStickButtonPressed(final int stick, final int button) {
     if (stick < 0 || stick >= kJoystickPorts) {
       throw new IllegalArgumentException("Joystick index is out of range, should be 0-3");
     }
-    boolean error = false;
-    boolean retVal = false;
-    synchronized (m_cacheDataMutex) {
+    m_cacheDataMutex.lock();
+    try {
       if (button > m_joystickButtons[stick].m_count) {
-        error = true;
-        retVal = false;
-      } else {
-        // If button was pressed, clear flag and return true
-        if ((m_joystickButtonsPressed[stick] & 1 << (button - 1)) != 0) {
-          m_joystickButtonsPressed[stick] &= ~(1 << (button - 1));
-          retVal = true;
-        } else {
-          retVal = false;
-        }
+        // Unlock early so error printing isn't locked.
+        m_cacheDataMutex.unlock();
+        reportJoystickUnpluggedWarning("Joystick Button " + button + " on port " + stick
+            + " not available, check if controller is plugged in");
+      }
+
+      // If button was pressed, clear flag and return true
+      if ((m_joystickButtonsPressed[stick] & 1 << (button - 1)) != 0) {
+        m_joystickButtonsPressed[stick] &= ~(1 << (button - 1));
+        return true;
+      }
+    } finally {
+      if (m_cacheDataMutex.isHeldByCurrentThread()) {
+        m_cacheDataMutex.unlock();
```

Comment:
This is a bit ugly...

Reply:
This is the same thing that the other button methods currently do.

Reply:
It does occur to me now thinking about it that if we change the above behaviour then this can get cleaned up.